arelle/plugin/txmyPkgAutoLoad.py

```diff
@@ -0,0 +1,47 @@
+"""
+
+Auto load taxonomy package for entry points in XBRL Taxonomy Registry (via STANDARD_PACKAGES_URL).
+
+See COPYRIGHT.md for copyright information.
+"""
+
+import json
+
+from arelle import PackageManager
+from arelle.DialogPackageManager import STANDARD_PACKAGES_URL
+from arelle.Version import authorLabel, copyrightLabel
+
+entryPointsPackageUrl = None
+
+def taxonomyPackageAutoLoad(_cntlr, packagesConfig, url):
+    global entryPointsPackageUrl
+    if packagesConfig is not None and url is not None and url.startswith("http"):
+        if entryPointsPackageUrl is None: # load standard taxonomy packages
+            entryPointsPackageUrl = {}
+            with open(_cntlr.webCache.getfilename(STANDARD_PACKAGES_URL, reload=True), 'r', errors='replace') as fh:
+                regPkgs = json.load(fh) # always reload
+            for pkgTxmy in regPkgs.get("taxonomies", []):
+                _url = pkgTxmy.get("Links",{}).get("AuthoritativeURL")
+                if _url.endswith(".zip"):
+                    # Ignore taxonomies that lack a direct download link.
+                    # Although future valid taxonomy download links from the registry might not have a ".zip" extension,
+                    # as of October 27th, 2024, all links that without a zip file extension have been invalid.
+                    for _ep in pkgTxmy.get("EntryPoints") or ():
+                        for _doc in _ep.get("EntryPointDocuments") or ():
+                            entryPointsPackageUrl[_doc] = _url
+        if url in entryPointsPackageUrl and not any(
+            url.startswith(mapFrom) and not url.startswith(mapTo)
```

Comment:
It seems like this check (does the package manager already have a matching package) would be better performed before the download and parsing steps above?

Reply:
The preceding "if entryPointsPackageUrl is None" loads entryPointsPackageUrl the first time (when used in a given invocation).  Otherwise "url in entryPointsPackageUrl" would be asking "url in None" before the registry is loaded.

Reply:
I'm thinking of the case of the GUI, where someone has this plugin enabled as a catch all, but they frequently already have the required packages. It will redownload the registry once for each instance of the GUI regardless of if they need it.

Are there any downsides to performing this check earlier?